Remove old payment gateway task components

joshuatf opened the pull request:
Fixes #7216 

* Removes old payment gateway task components
* Cleans up component folder organization
* Turns on the PaymentGatewaySuggestions feature in all environments

### Screenshots

<img width="717" alt="Screen Shot 2021-06-22 at 7 44 03 PM" src="https://user-images.githubusercontent.com/10561050/123013434-37377780-d392-11eb-8426-aa0d14d26f2a.png">


### Detailed test instructions:

1. Smoke test the payments task
2. Build the plugin zip and make sure the feature is enabled and working
3. Optionally, turn off the `payment-gateway-suggestions` feature flag and note that it does not appear in the task list

joshuatf committed:
Remove old payment gateway task components

joshuatf committed:
Move PaymentGatewaySuggestions up one directory

joshuatf committed:
Add feature check to tasks array

joshuatf committed:
Turn on payment gateway suggestions in all environments

louwie17 reviewed (commented):
There still appears to be some build errors with imports and usage of deleted files:
`getPaymentMethods` - client/header/activity-panel/panels/help.js
`isWCPaySupported` - ./client/payments/payment-recommendations.tsx
`WCPayUsageModal` - /client/layout/index.js
`isWCPaySupported` - ../selective-bundle/selective-extensions-bundle/index.js

joshuatf committed:
Handle PR feedback

joshuatf committed:
Use gateway suggestions from REST API in help panel

joshuatf commented:
Thanks for taking a look, @louwie17!  Fixed those references and updated the retrieval of gateway suggestions in https://github.com/woocommerce/woocommerce-admin/pull/7224/commits/a69e2ec0b406909e652704f196d2e780dedb8480.

joshuatf committed:
Fix component path

joshuatf committed:
Remove BACS test due to SlotFill testing bug

joshuatf commented:
@louwie17  This is ready for another look.  The failing test on this was the BACS setup task.  I believe this is due to the issue with SlotFill not working during tests.  Can you confirm @joelclimbsthings?

In non-headless mode, all tests pass, but in headless mode, that one test fails.  I hate to remove this test, but not quite sure how to get around this failure.

joshuatf committed:
Await button selector

louwie17 commented on client/header/activity-panel/panels/help.js:
```diff
@@ -123,53 +101,53 @@ function getPaymentsItems( props ) {
 				'https://docs.woocommerce.com/document/payfast-payment-gateway/?utm_source=help_panel',
 		},
-		showEway && {
+		paymentGatewaySuggestions.eway_payments && {
```
I assume this should be changed to `eway` in relation to this PR: https://github.com/woocommerce/woocommerce-admin/pull/7228

joshuatf replied:
Good catch!  Updated.

louwie17 reviewed (commented):
Nice work on this @joshuatf I am enjoying this very much:
<img width="150" alt="Screen Shot 2021-06-24 at 5 29 38 PM" src="https://user-images.githubusercontent.com/2240960/123328256-c1f2b080-d511-11eb-9291-55dfb2337a70.png">
It will feel nice to have most of the payment stuff pushed out onto the plugins versus our repo :)
Everything tested well, there is just one small change for the eway help item, oooh and don't forget to add a changelog for this 😄

joshuatf committed:
Handle PR feedback round 2

joshuatf commented:
Thanks for taking another look, @louwie17.  Addressed the `eway` change and rebased and added a changelog entry.

joelclimbsthings commented:
> This is ready for another look. The failing test on this was the BACS setup task. I believe this is due to the issue with SlotFill not working during tests. Can you confirm @joelclimbsthings?

Looks like tests are passing, so this is resolved?

louwie17 commented on readme.txt:
```diff
@@ -79,20 +79,21 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
 - Fix: The use of gridicons for Analytics section controls. #7237
-- Dev: Add `woocommerce_admin_export_id` filter for customizing the export file name #7178
+- Tweak: Revert Card component removal #7167
 - Update: Notes to use a date range. #7222
```
Thanks for re-arranging these, it be nice to keep them in alphabetical order as it prevents conflicts 😄

louwie17 reviewed (approved):
Thanks for the update @joshuatf the Eway gateway tested well this time 👍 👍
